File: src/main/java/valyush/platform/repository/CourseRepository.java
package valyush.platform.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import valyush.platform.entity.Course;

import java.util.List;

@Repository
public interface CourseRepository extends JpaRepository<Course, Long> {
    List<Course> findByCategoryId(Long categoryId);

    List<Course> findByTeacherId(Long teacherId);
}

File: src/main/kotlin/v/orm/domain/course/repository/CourseRepository.kt
package v.orm.domain.course.repository

import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import v.orm.domain.course.entity.Course
import java.util.*

interface CourseRepository : JpaRepository<Course, UUID> {

    fun findByCategory_Id(categoryId: UUID): List<Course>

    @EntityGraph(attributePaths = ["modules", "modules.lessons"])
    fun findDetailedById(id: UUID): Course?
}